lib/flipper/adapters/moneta.rb

```diff
@@ -0,0 +1,122 @@
+require 'moneta'
+
+module Flipper
+  module Adapters
+    class Moneta
+      include ::Flipper::Adapter
+
+      FEATURES_KEY = :flipper_features
+
+      # Public: The name of the adapter.
+      attr_reader :name
+
+      # Public
+      def initialize(moneta)
+        @moneta = moneta
+        @name = :moneta
+      end
+
+      # Public:  The set of known features
+      def features
+        moneta[FEATURES_KEY] || Set.new
+      end
+
+      # Public: Adds a feature to the set of known features.
+      def add(feature)
+        moneta[FEATURES_KEY] = features << feature.key.to_s
+        true
+      end
+
+      # Public: Removes a feature from the set of known features and clears
+      # all the values for the feature.
+      def remove(feature)
+        moneta[FEATURES_KEY] = features.delete(feature.key.to_s)
+        moneta[key(feature.key)] = default_config
+        true
+      end
+
+      # Public: Clears all the gate values for a feature.
+      def clear(feature)
+        moneta[key(feature.key)] = default_config
+        true
+      end
+
+      # Public: Gets the values for all gates for a given feature.
+      #
+      # Returns a Hash of Flipper::Gate#key => value.
+      def get(feature)
+        default_config.merge(moneta[key(feature.key)].to_h)
+      end
+
+      # Public
+      def get_multi(features)
+        result = {}
+        features.each do |feature|
+          result[feature.key] = get(feature)
+        end
+        result
+      end
+
+      # Public
+      def get_all
+        result = {}
+        moneta[FEATURES_KEY].each do |feature_key|
+          result[feature_key] = get(Feature.new(feature_key, self))
+        end
+        result
+      end
+
+      # Public: Enables a gate for a given thing.
+      #
+      # feature - The Flipper::Feature for the gate.
+      # gate - The Flipper::Gate to disable.
+      # thing - The Flipper::Type being enabled for the gate.
+      #
+      # Returns true.
+      def enable(feature, gate, thing)
+        case gate.data_type
+        when :boolean, :integer
+          result = get(feature)
+          result[gate.key] = thing.value.to_s
+          moneta[key(feature.key)] = result
+        when :set
+          result = get(feature)
+          result[gate.key] << thing.value.to_s
+          moneta[key(feature.key)] = result
```

Comment:
possible race condition here, but since the Moneta key points to a plain old ruby hash we need to update the hash in memory and then reset the key

Reply:
Yep, no problem. The KV branch I have going takes the same approach. Writes are rare so this seems totally fine to me. If people can't live with this, they can create a locking adapter to wrap moneta with some form of locking to avoid the race.